Idempotency + rolling upgrade sample

jonasvermeulen opened the pull request:
Updated the role to make it idempotent and split off generation of qcow and definition of vm in separate tasks file

Included sample playbook with serial: 1 to orchestrate rolling upgrade.

jonasvermeulen committed:
Updated the role to make it idempotent and split off generation of qcow and definition of vm in separate tasks file
Included sample playbook with serial:1 to ensure rolling upgrade.

jonasvermeulen committed:
Updated XMPP username/password setting as part of VSC config file

jonasvermeulen committed:
Change location of vsc.xml.j2 to templates directory

Commit:
Redeploy VSC VM when domain XML changes
Updated domain XML to disable CPU affinity in case of qemu domain type

hellt commented:
Hi *
any chances this PR will be merged here or on our gitlab?

hellt commented on tasks/definevscvm.yml:
```diff
@@ -0,0 +1,37 @@
+  delegate_to: "{{ hypervisor }}"
+  when: inventory_hostname in virt_vms.list_vms
+
+- name: Copy over VSC qcow2 image
```
It seems like another copying procedure. The first one is still in place in `main.yml`

jonasvermeulen replied:
The first copy is the "src" one.
This is to do a comparison if the binary has changed. This is the case when
you want to install a newer version. It is impossible to compare this on
the actual qcow since this has a changed config file inside.

The second copy will be the actual qcow where the config has to be embbeded
into.

A complete different alternative implementation is to install netmiko
library and request the version dynamically.

On Tue, May 24, 2016 at 9:42 PM, Roman Dodin notifications@github.com
wrote:

> In tasks/definevscvm.yml
> https://github.com/rvichery/nuage-vsc-installer/pull/6#discussion_r64456444
> :
> 
> > -  register: virt_vms
> > -  delegate_to: "{{ hypervisor }}"
> >   +
> >   +# Destroy if running VSC Guest VM
> >   +- name: Destroy VSC Guest VM if still running
> > -  virt: name={{ inventory_hostname }} state=destroyed
> > -  delegate_to: "{{ hypervisor }}"
> > -  when: inventory_hostname in virt_vms.list_vms
> >   +
> >   +# Undefine any previously existing VSC Guest VM
> >   +- name: Undefine VSC Guest VM if already provisioned
> > -  virt: name={{ inventory_hostname }} command=undefine
> > -  delegate_to: "{{ hypervisor }}"
> > -  when: inventory_hostname in virt_vms.list_vms
> >   +
> >   +- name: Copy over VSC qcow2 image
> 
> It seems like another copying procedure. The first one is still in place
> in main.yml
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/rvichery/nuage-vsc-installer/pull/6/files/51d3faa272440df5112ebcfc6f4eb001311e097e#r64456444

hellt replied:
Makes sense indeed.
I didn't test/try it, but maybe it would be quickier to get a sha256 (from `stat` module) from original image, store it in a file and later compare hashes of a new qcow2?

jonasvermeulen replied:
Could be good idea.
Feel free to do a fork of my code, and send me a PR !

On Wed, May 25, 2016 at 8:25 AM, Roman Dodin notifications@github.com
wrote:

> In tasks/definevscvm.yml
> https://github.com/rvichery/nuage-vsc-installer/pull/6#discussion_r64520988
> :
> 
> > -  register: virt_vms
> > -  delegate_to: "{{ hypervisor }}"
> >   +
> >   +# Destroy if running VSC Guest VM
> >   +- name: Destroy VSC Guest VM if still running
> > -  virt: name={{ inventory_hostname }} state=destroyed
> > -  delegate_to: "{{ hypervisor }}"
> > -  when: inventory_hostname in virt_vms.list_vms
> >   +
> >   +# Undefine any previously existing VSC Guest VM
> >   +- name: Undefine VSC Guest VM if already provisioned
> > -  virt: name={{ inventory_hostname }} command=undefine
> > -  delegate_to: "{{ hypervisor }}"
> > -  when: inventory_hostname in virt_vms.list_vms
> >   +
> >   +- name: Copy over VSC qcow2 image
> 
> Makes sense indeed.
> I didn't test/try it, but maybe it would be quickier to get a sha256 (from
> stat module) from original image, store it in a file and later compare
> hashes of a new qcow2?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/rvichery/nuage-vsc-installer/pull/6/files/51d3faa272440df5112ebcfc6f4eb001311e097e#r64520988

jonasvermeulen committed:
Add default static routes pointing to gateway IP as configured under mgmt and control interface variables
Add support for BGP peering via control IP (instead of system IP)

jonasvermeulen committed:
Add support for a default route to a gateway-ip in bof and main router table
Add support for qemu virtualized XML file
Added startup of libvirtd service as part of deployment tasks
Redeploy VSC VM if VM is not in 'running' state

jonasvermeulen committed:
Corrected CLI in config.cfg
